app.go

```diff
@@ -481,6 +499,14 @@ func (a *App) VisibleCommands() []*Command {
 	return ret
 }
 
+// VisibleFlagCategories returns a slice containing all the categories with the flags they contain
+func (a *App) VisibleFlagCategories() []VisibleFlagCategory {
+	if a.flagCategories == nil {
+		a.flagCategories = newFlagCategories()
```

Comment:
cant you return nil or [] if a.flagCategories is nil ? Its not nice behaviour to update flagCategories here

Reply:
I agree!